### ISSUE TITLE
JetStream cluster could not decode consumer snapshot

### ISSUE THREAD

**Issue by raoptimus** [nats-io/nats-server#2720]
## Defect

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [ ] Included `nats-server -DV` output
 - [ ] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats-server` and affected client libraries used:
2.6.5

#### OS/Container environment:
alpine 

#### Steps or code to reproduce the issue:
- stop 1 of 3 pods
- start 1 of 3 pods

#### Expected result:
- correct start

#### Actual result:
- incorrect start

```
kubectl -n production  logs collector-event-nats-2 -c nats
[87] 2021/12/01 17:08:48.861842 [INF] Starting nats-server
[87] 2021/12/01 17:08:48.861966 [INF]   Version:  2.6.5
[87] 2021/12/01 17:08:48.861971 [INF]   Git:      [ea48105]
[87] 2021/12/01 17:08:48.861974 [INF]   Name:     collector-event-nats-2
[87] 2021/12/01 17:08:48.861989 [INF]   Node:     dcRC6Zf4
[87] 2021/12/01 17:08:48.861992 [INF]   ID:       NBPSM6UJMEPOKF6FNJODGZX2YDULV3L7NVNK4RNXECA72OE467G27VDD
[87] 2021/12/01 17:08:48.861998 [INF] Using configuration file: /etc/nats-config/nats.conf
[87] 2021/12/01 17:08:48.863486 [INF] Starting JetStream
[87] 2021/12/01 17:08:48.863829 [INF]     _ ___ _____ ___ _____ ___ ___   _   __  __
[87] 2021/12/01 17:08:48.863846 [INF]  _ | | __|_   _/ __|_   _| _ \ __| /_\ |  \/  |
[87] 2021/12/01 17:08:48.863848 [INF] | || | _|  | | \__ \ | | |   / _| / _ \| |\/| |
[87] 2021/12/01 17:08:48.863850 [INF]  \__/|___| |_| |___/ |_| |_|_\___/_/ \_\_|  |_|
[87] 2021/12/01 17:08:48.863851 [INF] 
[87] 2021/12/01 17:08:48.863853 [INF]          https://docs.nats.io/jetstream
[87] 2021/12/01 17:08:48.863854 [INF] 
[87] 2021/12/01 17:08:48.863856 [INF] ---------------- JETSTREAM ----------------
[87] 2021/12/01 17:08:48.863869 [INF]   Max Memory:      512.00 MB
[87] 2021/12/01 17:08:48.863872 [INF]   Max Storage:     50.00 GB
[87] 2021/12/01 17:08:48.863875 [INF]   Store Directory: "/data/jetstream"
[87] 2021/12/01 17:08:48.863876 [INF] -------------------------------------------
[87] 2021/12/01 17:08:48.870153 [INF]   Restored 0 messages for stream "error-collector"
[87] 2021/12/01 17:08:56.299199 [INF]   Restored 5,019,337 messages for stream "event-collector"
[87] 2021/12/01 17:09:08.709647 [INF]   Restored 5,184,221 messages for stream "request-collector"
[87] 2021/12/01 17:09:08.709714 [INF]   Recovering 1 consumers for stream - "error-collector"
[87] 2021/12/01 17:09:08.710007 [INF]   Recovering 1 consumers for stream - "event-collector"
[87] 2021/12/01 17:09:08.710163 [INF]   Recovering 1 consumers for stream - "request-collector"
[87] 2021/12/01 17:09:08.710383 [INF] Starting JetStream cluster
[87] 2021/12/01 17:09:08.710387 [INF] Creating JetStream metadata controller
[87] 2021/12/01 17:09:08.710598 [INF] JetStream cluster recovering state
[87] 2021/12/01 17:09:08.712197 [INF] Starting http monitor on 0.0.0.0:8222
[87] 2021/12/01 17:09:08.712280 [INF] Listening for client connections on 0.0.0.0:4222
[87] 2021/12/01 17:09:08.712503 [INF] Server is ready
[87] 2021/12/01 17:09:08.712522 [INF] Cluster name is nats
[87] 2021/12/01 17:09:08.712548 [INF] Listening for route connections on 0.0.0.0:6222
[87] 2021/12/01 17:09:08.716585 [INF] 10.244.243.194:6222 - rid:22 - Route connection created
[87] 2021/12/01 17:09:08.716643 [INF] 10.244.69.177:6222 - rid:23 - Route connection created
[87] 2021/12/01 17:09:08.753549 [WRN] Error applying entries to '$G > event-collector': last sequence mismatch
[87] 2021/12/01 17:09:08.768254 [WRN] Resetting stream cluster state for '$G > event-collector'
[87] 2021/12/01 17:09:08.795096 [WRN] Error applying entries to '$G > event-collector': last sequence mismatch
[87] 2021/12/01 17:09:08.797944 [ERR] JetStream cluster could not decode consumer snapshot for '$G > request-collector > request-collector-1' [C-R3F-XO5nmQ6F]
panic: corrupt state file

goroutine 179 [running]:
github.com/nats-io/nats-server/server.(*jetStream).applyConsumerEntries(0xc000162160, 0xc000625500, 0xc00059e4c0, 0x0, 0x5, 0x1)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:3037 +0xe6f
github.com/nats-io/nats-server/server.(*jetStream).monitorConsumer(0xc000162160, 0xc000625500, 0xc00067eab0)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:3006 +0x68f
github.com/nats-io/nats-server/server.(*jetStream).processClusterCreateConsumer.func1()
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:2800 +0x3c
created by github.com/nats-io/nats-server/server.(*Server).startGoRoutine
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/server.go:2867 +0xc5

```

**Comment by derekcollison** [comment]
If you would not mind sending us that file that would be great. Then if you remove that whole directory the server should start.

The directory should be..

```/data/jetstream/$SYS/_js_/C-R3F-XO5nmQ6F```

**Comment by raoptimus** [comment]
Sorry I don't have the file. I have removed this directory. It did not help

**Comment by derekcollison** [comment]
ok so the file has been removed but the system has not repaired itself, that correct?

**Comment by raoptimus** [comment]
> 

yes. maybe need to remove it on everyone servers

**Comment by derekcollison** [comment]
Let us know if this issue persists, but we believe we got it with this fix. Will be in nightly builds and of course as part of next release.

**Comment by muralov** [comment]
We also faced the issue yesterday, with nats-server 2.9.6 version as it follows:
```
root@W-R90XW6BB:~# kubectl -n kyma-system logs eventing-nats-2 --all-containers
2022/12/14 19:29:32 Starting NATS Server Reloader v0.7.2
2022/12/14 19:29:32 Error: open /var/run/nats/nats.pid: no such file or directory
2022/12/14 19:29:34 found pid from pidfile "/var/run/nats/nats.pid" after 1 failed attempts (2.001139979s time after start)
2022/12/14 19:29:34 Live, ready to kick pid 46 (live, from 46 spec) based on any of 1 files
[20] 2022/12/14 19:30:02.708248 [ERR] Could not find server id: Get "http://localhost:8222//varz": dial tcp [::1]:8222: connect: connection refused
[20] 2022/12/14 19:30:32.708959 [ERR] Could not find server id: Get "http://localhost:8222//varz": dial tcp [::1]:8222: connect: connection refused
[20] 2022/12/14 19:31:02.708648 [ERR] Could not find server id: Get "http://localhost:8222//varz": dial tcp [::1]:8222: connect: connection refused
[20] 2022/12/14 19:31:32.708312 [ERR] Could not find server id: Get "http://localhost:8222//varz": dial tcp [::1]:8222: connect: connection refused
[20] 2022/12/14 19:32:02.709046 [ERR] Could not find server id: Get "http://localhost:8222//varz": dial tcp [::1]:8222: connect: connection refused
[20] 2022/12/14 19:32:32.708718 [ERR] Could not find server id: Get "http://localhost:8222//varz": dial tcp [::1]:8222: connect: connection refused
[20] 2022/12/14 19:33:02.708901 [ERR] Could not find server id: Get "http://localhost:8222//varz": dial tcp [::1]:8222: connect: connection refused
[111] 2022/12/14 19:32:36.441656 [INF] Starting nats-server
[111] 2022/12/14 19:32:36.441697 [INF]   Version:  2.9.6
[111] 2022/12/14 19:32:36.441700 [INF]   Git:      [289a9e1]
[111] 2022/12/14 19:32:36.441702 [INF]   Cluster:  eventing-nats
[111] 2022/12/14 19:32:36.441704 [INF]   Name:     eventing-nats-2
[111] 2022/12/14 19:32:36.441707 [INF]   Node:     LQ7xh9M7
[111] 2022/12/14 19:32:36.441709 [INF]   ID:       NBP6NIZHE37JLQ2JGLWYNL32ZGKHXK4QC6HTNNCFKUMYTKEXOQI4TXOP
[111] 2022/12/14 19:32:36.441714 [INF] Using configuration file: /etc/nats-config/nats.conf
[111] 2022/12/14 19:32:36.442228 [INF] Starting http monitor on 0.0.0.0:8222
[111] 2022/12/14 19:32:36.442271 [INF] Starting JetStream
[111] 2022/12/14 19:32:36.442345 [INF]     _ ___ _____ ___ _____ ___ ___   _   __  __
[111] 2022/12/14 19:32:36.442351 [INF]  _ | | __|_   _/ __|_   _| _ \ __| /_\ |  \/  |
[111] 2022/12/14 19:32:36.442353 [INF] | || | _|  | | \__ \ | | |   / _| / _ \| |\/| |
[111] 2022/12/14 19:32:36.442355 [INF]  \__/|___| |_| |___/ |_| |_|_\___/_/ \_\_|  |_|
[111] 2022/12/14 19:32:36.442356 [INF]
[111] 2022/12/14 19:32:36.442358 [INF]          https://docs.nats.io/jetstream
[111] 2022/12/14 19:32:36.442359 [INF]
[111] 2022/12/14 19:32:36.442361 [INF] ---------------- JETSTREAM ----------------
[111] 2022/12/14 19:32:36.442367 [INF]   Max Memory:      1.00 GB
[111] 2022/12/14 19:32:36.442370 [INF]   Max Storage:     1.00 GB
[111] 2022/12/14 19:32:36.442372 [INF]   Store Directory: "/data/jetstream"
[111] 2022/12/14 19:32:36.442373 [INF] -------------------------------------------
[111] 2022/12/14 19:32:36.442670 [INF]   Starting restore for stream '$G > sap'
[111] 2022/12/14 19:32:36.442964 [INF]   Restored 118 messages for stream '$G > sap'
[111] 2022/12/14 19:32:36.443005 [INF]   Recovering 3 consumers for stream - '$G > sap'
[111] 2022/12/14 19:32:36.443758 [INF] Starting JetStream cluster
[111] 2022/12/14 19:32:36.443768 [INF] Creating JetStream metadata controller
[111] 2022/12/14 19:32:36.443939 [INF] JetStream cluster recovering state
[111] 2022/12/14 19:32:36.444352 [INF] Listening for client connections on 0.0.0.0:4222
[111] 2022/12/14 19:32:36.444660 [INF] Server is ready
[111] 2022/12/14 19:32:36.444737 [INF] Cluster name is eventing-nats
[111] 2022/12/14 19:32:36.444845 [INF] Listening for route connections on 0.0.0.0:6222
[111] 2022/12/14 19:32:36.448066 [ERR] JetStream cluster could not decode consumer snapshot for '$G > sap > f89ed6570e445d8885976af142fa5393' [C-R3F-09t8WKrp]
panic: corrupt state file

goroutine 73 [running]:
github.com/nats-io/nats-server/v2/server.(*jetStream).applyConsumerEntries(0xc00026c000, 0xc00033ed80, 0x0?, 0x0)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:4002 +0x816
github.com/nats-io/nats-server/v2/server.(*jetStream).monitorConsumer(0xc00026c000, 0xc00033ed80, 0xc000150b40)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:3881 +0xf76
github.com/nats-io/nats-server/v2/server.(*jetStream).processClusterCreateConsumer.func1()
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:3591 +0x25
created by github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/server.go:3077 +0x85

```
The issue seems to appear with the nats-server 2.9.6 version again. Can you please have a look?

**Comment by derekcollison** [comment]
This has been fixed in latest server version 2.9.9.
